### ISSUE TITLE
Docstring errors in laf.py normalize/denormalize and an unused match_fginn argument

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4067]
### Describe the bug

Documentation-only defects found while auditing `kornia/feature`. No behaviour change involved.

#### `normalize_laf` / `denormalize_laf` docstrings — `kornia/feature/laf.py`

https://github.com/kornia/kornia/blob/main/kornia/feature/laf.py#L317-L334

1. **`:317`** — the summary line reads `"""De-F.normalize LAFs from scale to image scale.` The `F.` is a botched search-and-replace of `normalize` → `F.normalize` (the module imports `torch.nn.functional as F`). It should read *"Denormalize LAFs…"*.

2. **`:327`** — the worked example says the second row becomes

   ```
   [a21*MIN_SIZE a22*MIN_SIZE y*(W-1)]
   ```

   but the code multiplies the y translation by `h - 1`:

   ```python
   coef[0, 0, 0, 2] = wf   # w - 1
   coef[0, 0, 1, 2] = hf   # h - 1
   ```

   so it should be `y*(H-1)`. `normalize_laf` states the corresponding line correctly (`y/(H-1)`).

3. **`:323` and `:354`** — both docstrings write the source matrix as

   ```
   [a11 a21 x]
   [a21 a22 y]
   ```

   with `a21` in the top row. The top row is `[a11 a12 x]`.

4. **`:365`** — `normalize_laf`'s `Returns:` says *"the **denormalized** LAF … scale in image percentage (0, 1)"*. It returns the normalized LAF.

#### `match_fginn` accepts an unused argument — `kornia/feature/matching.py`

https://github.com/kornia/kornia/blob/main/kornia/feature/matching.py#L257

`lafs1` is a required positional parameter and is documented as `:math:`(1, B1, 2, 3)``, but the function body never reads it — only `lafs2` is used, via `xy2 = get_laf_center(lafs2)`. Either it should be used, or the docstring should record that it is accepted for API symmetry with `match_adalam` and ignored.

### Expected behavior

Docstrings that match the code.

Found during an audit of `kornia.feature`.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by ducha-aiki** [comment]
One more of the same class, in `kornia/feature/siftdesc.py` — `SIFTDescriptor`'s documented input shape is wrong, and following it raises:

https://github.com/kornia/kornia/blob/main/kornia/feature/siftdesc.py#L104-L106

```
Shape:
    - Input: :math:`(B, 1, \text{num_spatial_bins}, \text{num_spatial_bins})`
    - Output: :math:`(B, \text{num_ang_bins * num_spatial_bins ** 2})`
```

`forward` requires `(B, 1, patch_size, patch_size)`:

```python
KORNIA_CHECK_SHAPE(input, ["B", "1", f"{self.patch_size}", f"{self.patch_size}"])
```

```python
import torch
from kornia.feature import SIFTDescriptor

s = SIFTDescriptor(patch_size=32, num_spatial_bins=4)
s(torch.rand(2, 1, 4, 4))        # the documented shape
```

```
Shape mismatch at dimension 2: expected 32, got 4.
  Expected shape: ['B', '1', '32', '32']
```

The input shape should read `(B, 1, \text{patch\_size}, \text{patch\_size})`. The class docstring's `Returns:` line is also truncated mid-sentence — *"SIFT descriptor of the patches with shape."*

Separately, `get_sift_bin_ksize_stride_pad` raises with mangled whitespace from a broken implicit string concatenation, which reaches users verbatim:

https://github.com/kornia/kornia/blob/main/kornia/feature/siftdesc.py#L83-L87

```
Patch size 16 is incompatible with             requested number of spatial bins 4             for SIFT descriptor. Usually it happens when patch size is too small            for num_spatial_bins specified
```

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by ducha-aiki** [comment]
**Scope update — 5 of the 6 items in this issue have landed. One is left, and it is a one-line docstring change.**

Verified against `main` at `d2166cfa`.

## Landed

| # | Item | Landed in |
| --- | --- | --- |
| 1 | `"""De-F.normalize LAFs…` | #4072 — now reads *"Denormalize LAFs from the [0, 1] scale to image (pixel) scale."* |
| 2 | `y*(W-1)` in `denormalize_laf`'s worked example | #4072 — now `y*(H-1)`, matching `coef[0, 0, 1, 2] = hf` |
| 3 | `[a11 a21 x]` top row in both docstrings | #4072 — now `[a11 a12 x]` in `denormalize_laf` and `normalize_laf` |
| 4 | `normalize_laf` `Returns:` saying *"denormalized"* | #4072 — now *"the normalized LAF"* |
| — | `SIFTDescriptor` documented input shape (raised in the comment above) | #4072 — `Shape: - Input: (B, 1, patch_size, patch_size)` |

`get_sift_bin_ksize_stride_pad`'s `ggiven` typo, adjacent to the same block, was fixed separately in #4077 by @Sid-5137.

## Still open — the whole remaining scope

`match_fginn`'s `lafs1` is still a required positional parameter that the body never reads. On `main`:

```console
$ git show origin/main:kornia/feature/matching.py \
    | sed -n '/^def match_fginn/,/^    KORNIA_CHECK_SHAPE(desc1/p' | grep -n lafs1
4:    lafs1: torch.Tensor,
29:        lafs1: LAFs of a shape :math:`(1, B1, 2, 3)`.
```

Only `lafs2` is consumed, via `xy2 = get_laf_center(lafs2)`.

**The fix here is to document it, not to remove it.** `lafs1` cannot be dropped: `GeometryAwareDescriptorMatcher.forward` dispatches every geometry-aware matcher through the same 4-argument call, so `match_fginn`'s signature has to stay symmetric with `match_adalam`'s. I raised removing it on #4075 and retracted it for that reason.

So the remaining change is to `kornia/feature/matching.py`, in `match_fginn`'s `Args:` block — extend the `lafs1` line to record that it is accepted for API symmetry with `match_adalam` and is not used by this function. No behaviour change, no signature change.

Still labelled `good first issue` / `Difficulty: easy`, and now genuinely small. **@alinaqvi1129** — if you are looking for something to pick up, this one is unclaimed and is a good fit; say the word and I will assign it to you.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by alinaqvi1129** [comment]
I’d be happy to take this one. Please assign it to me. Thanks!

**Comment by ducha-aiki** [comment]
@alinaqvi1129 — yours (see also #4096, which I have just given you as well). Only the last item is left: the `lafs1` argument of `match_fginn` is required by `GeometryAwareDescriptorMatcher.forward`, so the docstring should say *why* the argument exists rather than remove it. One-line change; fold it into your #4096 PR or open it separately, either is fine. Non-collaborators cannot be assigned on this repo, so this comment is the claim.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
